.changeset/gentle-windows-enjoy.md

```diff
@@ -0,0 +1,46 @@
+---
+"@astrojs/react": minor
+"astro": minor
+---
+
+Add support for [the React 19 `useActionState()` hook](https://react.dev/reference/react/useActionState) when using Astro Actions. This introduces progressive enhancement when calling an Action with the `withState()` utility.
```

Comment:
```suggestion
Adds two new functions `experimental_getActionState()` and `experimental_withState()` to support [the React 19 `useActionState()` hook](https://react.dev/reference/react/useActionState) when using Astro Actions. This introduces progressive enhancement when calling an Action with the `withState()` utility.
```
Suggesting that we explicitly mention what the actual adds to `astro:actions` are!